### PR TITLE
Remove -O2 and -DNDEBUG from libMesh builds with script

### LINKED ISSUE #15120: Avoid passing -O2 and -DNDEBUG to libMesh builds
## Bug Description
Currently, libMesh will not throw away flags like `-O2` and `-DNDEBUG` when they are set in `$CPPFLAGS` and `$CXXFLAGS`. See https://github.com/libMesh/TIMPI/pull/28.

Our test debug PETSc alt test builds libMesh using conda, which has the following flags at runtime:
```
(order_error) [Ubuntu-16][/raid/harblh/moose]> echo $CPPFLAGS
-DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /raid/harblh/miniconda3/envs/order_error/include
(order_error) [Ubuntu-16][/raid/harblh/moose]> echo $CXXFLAGS
-fvisibility-inlines-hidden -std=c++17 -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /raid/harblh/miniconda3/envs/order_error/include
```

`-O2` and `-DNDEBUG` are getting passed to libMesh dbg compiles from this, causing failures.

This is simply a temp fix until libMesh is patched.

## Steps to Reproduce
Build libMesh dbg with the `scripts/update_and_rebuild_libmesh.sh` script using the conda environment from `conda install moose-petsc and run moose test.

## Impact
Debug libMesh build in this fashion has both `DEBUG` and `NDEBUG` defined! The fix is ugly, temporary, and allows next to pass.

### PR DESCRIPTION
**Pull request by loganharbour**
Closes #15120 

Also has the contents from #15113  in it so we can test quickly

### CODE REVIEW

**Comment by loganharbour** [comment]
Looks good. Test debug petsc alt just passed on rod with the same conda config.

File: scripts/update_and_rebuild_libmesh.sh [hunk 1]
```diff
@@ -141,6 +141,15 @@ if [ -z "$go_fast" ]; then
...
 
+  # This is a temprorary fix, see #15120
+  if [[ -n "$CPPFLAGS" ]]; then
+    export CPPFLAGS=${CPPFLAGS//-DNDEBUG/}
+    export CPPFLAGS=${CPPFLAGS//-O2/}
+  fi
+  if [[ -n "$CXXFLAGS" ]]; then
+    export CXXFLAGS=${CXXFLAGS//-O2/}
+  fi
+
```
Comment (roystgnr) [inline scripts/update_and_rebuild_libmesh.sh:152]: Can't speak to the other changes, but this looks good.

Comment (loganharbour) [reply scripts/update_and_rebuild_libmesh.sh:152]: Other changes are from things already in next that I wanted to test with this.

**Review by milljm — APPROVED** [review]
I tested the substitution on my end, and it removes the offending flags. 👍

**Pull request merged**